### ISSUE TITLE
[Feature Request]: several points to increase user-friendly approach

### ISSUE THREAD

**Issue by PetBuj** [jasp-stats/jasp-issues#1549]
### Description

I supposed to add several features to existing JASP-methods to make it more  useful.

### Purpose

The purpose is to incerase usability of JASP.

### Is your feature request related to a problem?

no

### Describe the solution you would like

1. add feature to PCA/EFA to enable store scores into data-sheet, 2. enable to input data for contingency tables(tests) by existing counts, 3. Add to Casewise diagnostics (Lin.regression) Jack-knife residuals and leverage, 4. Nonlinear regression.

### Describe alternatives that you have considered

I believe, all these features/methods are available in R packages and therefore, the process is highly possible.

### Additional context

Thank authors of JASP to work on this very practical free package!

**Comment by juliuspf** [comment]
Hi @PetBuj, thanks for the reports. 
- We are already tracking 1) under a different issue (https://github.com/jasp-stats/jasp-issues/issues/1666).
- 4) what do you mean by this? there is logistic regression in JASP...

**Comment by PetBuj** [comment]
Hello,

I mean something like this:


Minimum Iteration Section

Iteration            No. of         Percent of        Bar Chart

No.                   Clusters     Variation          of Percent

1                       2                  56.39               |||||||||||||||||

4                       3                  33.06               ||||||||||

7                       4                  25.69               ||||||||

(NCSS report)

Where PercentOfVariation means how much (0-100%) the selected clusters (2-k) in k-means are possible reduce variability of the objects (when covering by k-clusters).

Best regards,

P. Bujok

**Comment by juliuspf** [comment]
what do you think @Kucharssim ?

**Comment by Kucharssim** [comment]
Dear @PetBuj,

Thanks for the points for improvement, I will break it down in separate points.

1.  add feature to PCA/EFA to enable store scores into data-sheet:

As @juliuspf wrote, this is already covered by a different issue: https://github.com/jasp-stats/jasp-issues/issues/1666

2. enable to input data for contingency tables(tests) by existing counts:

I am not sure what you mean, the contingency analysis already allows specifying input data by using existing counts. Could you please clarify what is missing?

3. add to Casewise diagnostics (Lin.regression) Jack-knife residuals and leverage:

I think we could do that, perhaps @fqixiang could have a look...

4. nonlinear regression. As @juliuspf wrote, it's not clear what kind of analysis you mean by the term "nonlinear regression". Could you clarify, please?

------

Further, you wrote:
> I mean something like this:
>
>
>Minimum Iteration Section
>
>Iteration            No. of         Percent of        Bar Chart
>
>No.                   Clusters     Variation          of Percent
>
>1                       2                  56.39               |||||||||||||||||
>
>4                       3                  33.06               ||||||||||
>
>7                       4                  25.69               ||||||||
>
>(NCSS report)
>
>Where PercentOfVariation means how much (0-100%) the selected clusters (2-k) in k-means are possible reduce variability of the objects (when covering by k-clusters).

That seems to me that you meant to write this response to this issue instead https://github.com/jasp-stats/jasp-issues/issues/1407, is that correct? Thanks!

**Comment by PetBuj** [comment]
Dear Simon,

#1 – thanks
#2 - it is my fault I missed this possibility.
#3 – Thank for your effort
#4 – Nonlinear regression is ‚standard‘ procedure beside linear regression where y is dependent on Xs by a specific (no linear) model. The model contains one or more parameters which are estimated during the procedure (similarly to lin.reg but in diff.way). Some motivation is for example here (https://techvidvan.com/tutorials/nonlinear-regression-in-r/)

Best regards,

P. Bujok

**Comment by TarandeepKang** [comment]
Hi @Kucharssim & @juliuspfadt  For a grounding in nonlinear regression see:

https://link.springer.com/book/10.1007/978-0-387-09616-2

and a great implementation:

Florent Baty, Christian Ritz, Sandrine Charles, Martin Brutsche, Jean-Pierre Flandrois, Marie-Laure Delignette-Muller (2015). “A Toolbox for Nonlinear Regression in R: The Package nlstools.” Journal of Statistical Software, 66(5), 1–21. [doi:10.18637/jss.v066.i05](https://doi.org/10.18637/jss.v066.i05).

I was just coming to suggest this, but someone beat me to it!

**Comment by tomtomme** [comment]
1) Now tracked here: https://github.com/jasp-stats/jasp-issues/issues/101

2) done

3) Missing: Casewise diagnostics (Lin.regression) Jack-knife residuals and leverage

4) nonlinear regression: @TarandeepKang & @PetBuj - can you have a look at dustins explanations here:
https://github.com/jasp-stats/jasp-issues/issues/172
https://github.com/jasp-stats/jasp-issues/issues/2486
Is his implementation of non-linear Regression in the visual modeling module sufficient?

**Comment by tomtomme** [comment]
@TarandeepKang 
Can you have a look at point 3 from my comment above?
Is that implementation of non-linear regression sufficient? Or do we need nls from stats package?

**Comment by TarandeepKang** [comment]
So I would suggest that in the meantime we should find some way (maybe the help file) in the regression module of saying that polynomial regression is possible with flexplot. I had no idea! In the longer term, would be great to have maximum flexibility from nls, there will obviously be much work and I think there are greater priorities. Don't you agree?

**Comment by tomtomme** [comment]
Agreed. Added the documentation-hint to the list: https://github.com/jasp-stats/jasp-issues/issues/2529

remaining parts from this issue: 
adding 2 more casewise diagnostics for linear regression

- [ ]  Jack-knife residuals
- [ ]  leverage
- [ ] nls from stats package (longer term)

**Comment by PetBuj** [comment]
Yes, I agree.

Petr B
